pkg/router/router_test.go

```diff
@@ -86,13 +89,17 @@ func TestMain(m *testing.M) {
 
 	var plugin router.Plugin
 
-	workdir, err := ioutil.TempDir("", "router")
+	var err error
+	workdir, err = ioutil.TempDir("", "router")
 	if err != nil {
 		fmt.Println(err)
 		os.Exit(1)
 	}
 	fmt.Printf("router working directory: %s\n", workdir)
 
+	// directory for whitelist files (best effort, to not see error logs)
```

Comment:
You may need to update.  Is this change purposeful?

Reply:
It's just an error message I didn't notice when I first implemented the fix for the IP whitelists.
The error is about a missing directory for the whitelist files (long IP lists are put into a file as HAProxy config has a hard coded limitation of the length of config args).

I guess it brings a "better looking test" output.

Reply:
Changed this part a bit by adding `createRouteDirs` function.